src/ripple/app/tx/impl/PayChan.cpp

```diff
@@ -220,31 +234,50 @@ PayChanCreate::doApply()
     // Create PayChan in ledger
     auto const slep = std::make_shared<SLE> (
         keylet::payChan (account, dst, (*sle)[sfSequence] - 1));
-    // Funds held in this channel
-    (*slep)[sfAmount] = ctx_.tx[sfAmount];
-    // Amount channel has already paid
-    (*slep)[sfBalance] = ctx_.tx[sfAmount].zeroed ();
-    (*slep)[sfAccount] = account;
-    (*slep)[sfDestination] = dst;
-    (*slep)[sfSettleDelay] = ctx_.tx[sfSettleDelay];
-    (*slep)[sfPublicKey] = ctx_.tx[sfPublicKey];
-    (*slep)[~sfCancelAfter] = ctx_.tx[~sfCancelAfter];
-    (*slep)[~sfSourceTag] = ctx_.tx[~sfSourceTag];
-    (*slep)[~sfDestinationTag] = ctx_.tx[~sfDestinationTag];
-
-    ctx_.view ().insert (slep);
 
-    // Add PayChan to owner directory
+    STArray chanMembers {sfChannelMembers, 2};
+    chanMembers.push_back (STObject (sfChannelMember));
+    STObject& member1 = chanMembers.back ();
+    member1[sfBalance] = ctx_.tx[sfAmount];
+    member1[sfAccount] = account;
+    member1[sfPublicKey] = ctx_.tx[sfPublicKey];
+    member1[sfAmount] = XRPAmount{beast::zero};
+    member1[sfSequence] = 0;
+
+    chanMembers.push_back (STObject (sfChannelMember));
+    STObject& member2 = chanMembers.back ();
+    member2[sfBalance] = XRPAmount{beast::zero};
+    member2[sfAccount] = dst;
+    member2[sfPublicKey] = ctx_.tx[sfDstPublicKey];
+    member2[sfAmount] = XRPAmount{beast::zero};
+    member2[sfSequence] = 0;
+
+    // Add PayChan to owner directories
     {
         uint64_t page;
         auto result = dirAdd (ctx_.view (), page, keylet::ownerDir (account),
             slep->key (), describeOwnerDir (account),
             ctx_.app.journal ("View"));
         if (!isTesSuccess (result.first))
             return result.first;
-        (*slep)[sfOwnerNode] = page;
+        member1[sfOwnerNode] = page;
+
+        auto const sled = ctx_.view ().peek (keylet::account (dst));
+        result = dirAdd (ctx_.view (), page, keylet::ownerDir (dst),
+            slep->key (), describeOwnerDir (dst),
+            ctx_.app.journal ("View"));
+        if (!isTesSuccess (result.first))
+            return result.first;
+        member2[sfOwnerNode] = page;
+        ctx_.view ().update (sled);
```

Comment:
Removed extraneous `peek` and added lambda https://github.com/ripple/rippled/pull/1869/commits/8a5e7f6e49808a499e99b8288361174bc791125b